File: Python/58.Length of Last Word.py
#返回最后一个空格前的字符串长度(最后的空格不算)
class Solution(object):
    def lengthOfLastWord(self, s):
        l = len(s)
        if l==0:
            return 0
        i = l-1
        while i>0 and s[i]==' ':
            i -= 1
        lastnotempty = i
        while i>=0 and s[i]!=' ':
            i -= 1
        return lastnotempty-i

s = Solution()
print(s.lengthOfLastWord('a abvc'))

File: C++/String/58.Length of Last Word.cpp
/*
返回字符串中最后一个词的长度。不存在的话，返回0。
字符串只包含大小写字母和空格。
*/
#include <iostream>
#include <string>
using namespace std;

class Solution {
public:
	int lengthOfLastWord(string s) {
		int lastlen = 0,i;
		for(i=s.length()-1; i>=0 && s[i]==' '; --i);
        for(; i >= 0; --i)
        {
        	if(s[i]==' ')	break;
        	++lastlen;
        }
        return lastlen;
    }
};

int main()
{
	Solution s;;
	cout<<s.lengthOfLastWord("a ")<<endl;
}

